tests/Microsoft.Data.Analysis.Tests/DataFrameTests.cs

```diff
@@ -203,6 +203,23 @@ public void TestIndexer()
             Assert.Throws<ArgumentException>(() => dataFrame["Int5"]);
         }
 
+        [Fact]
+        public void GetColumnTests()
+        {
+            DataFrame dataFrame = MakeDataFrameWithAllColumnTypes(10);
+            PrimitiveDataFrameColumn<int> ints = dataFrame.Columns.GetColumn<PrimitiveDataFrameColumn<int>>("Int");
```

Comment:
I wonder if this API wouldn't be better:

```C#
PrimitiveDataFrameColumn<int> ints = dataFrame.Columns.GetPrimitiveColumn<int>("Int");
```

Reply:
That was my first idea. I wasn't sure if we were ok with adding new APIs for every new column type we come up with? I would be ok with introducing `GetPrimitiveColumn`, `GetStringColumn` and `GetArrowStringColumn`.

Reply:
For new column types that don't ship inside the same assembly, they can add extension methods.

From a callers stand-point, I would rather write `dataFrame.Columns.GetPrimitiveColumn<int>` vs `dataFrame.Columns.GetColumn<PrimitiveDataFrameColumn<int>>`. For one - there isn't a need for a nested generic type and two it is shorter.